ibis/expr/operations.py

```diff
@@ -2454,7 +2454,7 @@ class ExtractMillisecond(ExtractTimeField):
 
 class DayOfWeekIndex(UnaryOp):
     arg = Arg(rlz.one_of([rlz.date, rlz.timestamp]))
-    output_type = rlz.shape_like('arg', dt.int32)
+    output_type = rlz.shape_like('arg', dt.int16)
```

Comment:
Should that be `dt.int8`?

Reply:
it should, but neither `int8` nor `uint8` are supported across all databases. int16 is the smallest supported type

Reply:
I see! Later We might handle that at datatype level.

Reply:
Yeah, I think at some point we may want to have output types that can be determined by the backend.